src/shell/pipe_exec/mod.rs

```diff
@@ -431,8 +432,12 @@ impl PipelineExecution for Shell {
         };
 
         // If the given pipeline is a background task, fork the shell.
-        if let Some(command_name) = possible_background_name {
-            fork_pipe(self, piped_commands, command_name)
+        if let Some((command_name, disown)) = possible_background_name {
+            fork_pipe(self, piped_commands, command_name, if disown {
+                ProcessState::Empty
```

Comment:
I have no idea how to disown, I didn't understand the `disown` command's code a lot. So I did this.

Reply:
In this case, you don't need to add the job to the background process table at all. Slots that are marked as `Empty` will be rewritten when a new job is added to the table.

Reply:
Sorry for being slow. I got busy with other things. I'll look into it tomorrow!

Reply:
Effectively what you'll do is just have the parent do nothing after the fork -- the child will run in the background and the parent will have no knowledge of it.

Reply:
I just ended up conditionally running `send_to_background` in `fork_pipe`. Don't know if that's what you meant, sorry if not.